EXOAcceptedDomain: MatchSubDomains ValidateScript fails

#### Details of the scenario you tried and the problem that is occurring
I have an EXO configuration that I am trying to test with DSC (Test-DscConfiguration).
When I run the test I get an error message for the EXOAcceptedDomains objects that I have.
Example of EXOAcceptedDomain that passes the test without error : 
```powershell
        EXOAcceptedDomain [someID]
        {
            ApplicationId         = $ConfigurationData.NonNodeData.ApplicationId;
            CertificateThumbprint = $ConfigurationData.NonNodeData.CertificateThumbprint;
            DomainType            = "InternalRelay";
            Ensure                = "Present";
            Identity              = "somedomain.fr";
            MatchSubDomains       = $False;
            OutboundOnly          = $False;
            TenantId              = $ConfigurationData.NonNodeData.TenantId;
        }
```
Example that fails : 
```powershell
       EXOAcceptedDomain [someID]
        {
            ApplicationId         = $ConfigurationData.NonNodeData.ApplicationId;
            CertificateThumbprint = $ConfigurationData.NonNodeData.CertificateThumbprint;
            DomainType            = "InternalRelay";
            Ensure                = "Present";
            Identity              = "somedomain.fr";
            MatchSubDomains       = $True;
            OutboundOnly          = $False;
            TenantId              = $ConfigurationData.NonNodeData.TenantId;
        }
```

#### Verbose logs showing the problem
Log generated when I have the error : 
##[error]PowerShell DSC resource MSFT_EXOAcceptedDomain  failed to execute Test-TargetResource functionality with error 
message: Cannot validate argument on parameter 'MatchSubDomains'. The " $false -eq $_ " validation script for the 
argument with value "True" did not return a result of True. Determine why the validation script failed, and then try 
the command again. 

#### Suggested solution to the issue
I looked into MSFT_EXOAcceptedDomain.psm1 and it looks like there a difference between what's documented and the actual behavior.
The documentation says : 
The MatchSubDomains parameter must be false on Authoritative domains. The default value is false.
https://microsoft365dsc.com/resources/exchange/EXOAcceptedDomain/

The ValideScript doesn't take the "DomainType" into account.
It just says : 
```powershell
        [Parameter()]
        [ValidateScript( { $false -eq $_ })]
        [System.Boolean]
        $MatchSubDomains = $false,
```
I think one way to fix it would be to remove the ValidateScript completely and test the value in regards to the DomainType in the processing part of the Test-TargetResource function.

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
       EXOAcceptedDomain [someID]
        {
            ApplicationId         = $ConfigurationData.NonNodeData.ApplicationId;
            CertificateThumbprint = $ConfigurationData.NonNodeData.CertificateThumbprint;
            DomainType            = "InternalRelay";
            Ensure                = "Present";
            Identity              = "somedomain.fr";
            MatchSubDomains       = $True;
            OutboundOnly          = $False;
            TenantId              = $ConfigurationData.NonNodeData.TenantId;
        }
```

#### The operating system the target node is running
Windows 10

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.22.209.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,11 @@
 
 # 1.22.223.1
 
+* EXOAcceptedDomain
+  * Fixes an issue where True was never accepted as a value for MatchSubDomains
+    or OutboundOnly.
+    FIXES #1779
+
 * DEPENDENCIES
   * Updated DSCParser to version 1.3.0.3.
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,8 +3,8 @@
 # 1.22.223.1
 
 * EXOAcceptedDomain
-  * Fixes an issue where True was never accepted as a value for MatchSubDomains
-    or OutboundOnly.
+  * Fixes an issue where True was never accepted as a value for parameters
+    MatchSubDomains or OutboundOnly.
     FIXES #1779
 
 * DEPENDENCIES

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOAcceptedDomain/MSFT_EXOAcceptedDomain.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_EXOAcceptedDomain/MSFT_EXOAcceptedDomain.psm1
@@ -20,12 +20,10 @@ function Get-TargetResource
         $Ensure = 'Present',
 
         [Parameter()]
-        [ValidateScript( { $false -eq $_ })]
         [System.Boolean]
         $MatchSubDomains = $false,
 
         [Parameter()]
-        [ValidateScript( { $false -eq $_ })]
         [System.Boolean]
         $OutboundOnly = $false,
 
@@ -73,7 +71,7 @@ function Get-TargetResource
 
     #region Telemetry
     $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
-    $CommandName  = $MyInvocation.MyCommand
+    $CommandName = $MyInvocation.MyCommand
     $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
         -CommandName $CommandName `
         -Parameters $PSBoundParameters
@@ -203,12 +201,22 @@ function Set-TargetResource
 
     #region Telemetry
     $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
-    $CommandName  = $MyInvocation.MyCommand
+    $CommandName = $MyInvocation.MyCommand
     $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
         -CommandName $CommandName `
         -Parameters $PSBoundParameters
     Add-M365DSCTelemetryEvent -Data $data
     #endregion
+
+    if ($MatchSubDomains -eq $true -and $DomainType -ne "InternalRelay")
+    {
+        throw "MatchSubDomains can only be enabled on an InternalRelay domain."
+    }
+
+    if ($OutboundOnly -eq $true -and $DomainType -eq "ExternalRelay")
+    {
+        throw "OutboundOnly can only be enabled if the DomainType parameter is set to Authoritative or InternalRelay."
+    }
 
     $ConnectionMode = New-M365DSCConnection -Workload 'ExchangeOnline' `
         -InboundParameters $PSBoundParameters
@@ -246,12 +254,10 @@ function Test-TargetResource
         $Ensure = 'Present',
 
         [Parameter()]
-        [ValidateScript( { $false -eq $_ })]
         [System.Boolean]
         $MatchSubDomains = $false,
 
         [Parameter()]
-        [ValidateScript( { $false -eq $_ })]
         [System.Boolean]
         $OutboundOnly = $false,
 
@@ -284,7 +290,7 @@ function Test-TargetResource
 
     #region Telemetry
     $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
-    $CommandName  = $MyInvocation.MyCommand
+    $CommandName = $MyInvocation.MyCommand
     $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
         -CommandName $CommandName `
         -Parameters $PSBoundParameters
@@ -350,7 +356,7 @@ function Export-TargetResource
 
     #region Telemetry
     $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
-    $CommandName  = $MyInvocation.MyCommand
+    $CommandName = $MyInvocation.MyCommand
     $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
         -CommandName $CommandName `
         -Parameters $PSBoundParameters

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOAcceptedDomain/MSFT_EXOAcceptedDomain.schema.mof
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_EXOAcceptedDomain/MSFT_EXOAcceptedDomain.schema.mof
@@ -5,7 +5,7 @@ class MSFT_EXOAcceptedDomain : OMI_BaseResource
     [Write, Description("Specify if the AcceptedDomain should exist or not."), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] String Ensure;
     [Write, Description("The type of AcceptedDomain.  Currently the EXOAcceptedDomain DSC Resource accepts a value of 'Authoritative' and 'InternalRelay'."), ValueMap{"Authoritative","InternalRelay"}, Values{"Authoritative","InternalRelay"}] String DomainType;
     [Write, Description("The MatchSubDomains parameter must be false on Authoritative domains. The default value is false.")] Boolean MatchSubDomains;
-    [Write, Description("The OutboundOnly must be false on Authoritative domains. The default value is false.")] Boolean OutboundOnly;
+    [Write, Description("OutboundOnly can only be enabled if the DomainType parameter is set to Authoritative or InternalRelay. The default value is false.")] Boolean OutboundOnly;
     [Write, Description("Credentials of the Exchange Global Admin"), EmbeddedInstance("MSFT_Credential")] string Credential;
     [Write, Description("Id of the Azure Active Directory application to authenticate with.")] String ApplicationId;
     [Write, Description("Id of the Azure Active Directory tenant used for authentication.")] String TenantId;

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOAcceptedDomain/readme.md
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_EXOAcceptedDomain/readme.md
@@ -32,8 +32,7 @@ MatchSubDomains
 - Required: No
 - Description: MatchSubDomains enables mail to be sent by and received
   from users on any subdomain of this accepted domain.
-  This value must be false on Authoritative domains.
-  The EXOAcceptedDomain DSC Resource only accepts a value of $false
+  MatchSubDomains can only be enabled on an InternalRelay domain.
   The default value is false.
 
 OutboundOnly
@@ -42,8 +41,7 @@ OutboundOnly
 - Description: OutboundOnly specifies whether this accepted domain is an
   internal relay domain for the on-premises
   deployment for organizations that have coexistence with a cloud-based organization.
-  This value must be false on Authoritative domains.
-  The EXOAcceptedDomain DSC Resource only accepts a value of $false
+  OutboundOnly can only be enabled if the DomainType parameter is set to Authoritative or InternalRelay.
   The default value is false.
 
 ## Example
@@ -52,6 +50,8 @@ OutboundOnly
 EXOAcceptedDomain ExampleEmailDomain {
     Ensure              = 'Present'
     Identity            = 'example.contoso.com'
+    MatchSubDomains     = $false
+    OutboundOnly        = $false
     Credential          = $Credential
 }
 ```

